### PR TITLE
[ENH] Nicer reporting of tabular data (e.g. Confusion Matrix)

### PR DESCRIPTION
**Pull request by kernc**
Original title: "Fix report for Confusion Matrix"

### CODE REVIEW

File: Orange/canvas/report/report.py [hunk 1]
```diff
@@ -263,17 +279,10 @@ def report_list(data,
             ) for rowi, row in enumerate(data))
 
         self.report_name(name)
-        if isinstance(table, QAbstractItemModel):
-            model = table
-        else:
-            try:
-                model = table.model()
-            except:
-                model = None
```
Comment (kernc) [inline Orange/canvas/report/report.py]: This is not how you code.

File: Orange/canvas/report/report.py [hunk 2]
```diff
@@ -263,17 +279,10 @@ def report_list(data,
...
-        if isinstance(model, QStandardItemModel):
-            body = report_standard_model(model)
-        elif isinstance(model, QAbstractItemModel):
-            body = report_abstract_model(model, table)
+        if isinstance(table, QAbstractItemView):
+            body = report_abstract_model(table.model(), table)
+        elif isinstance(table, QAbstractItemModel):
+            body = report_abstract_model(table)
```
Comment (kernc) [inline Orange/canvas/report/report.py]: This is.

**Comment by astaric** [comment]
Does the new code work? Could you add some tests to prove it?

**Comment by kernc** [comment]
FTFY. These tests _pass_. All is ok. These are the tests you're looking for. :alien:

File: Orange/canvas/report/report.py [hunk 3]
```diff
@@ -195,54 +195,71 @@ def report_table(self, name, table=None, header_rows=0, header_columns=0,
...
+                bgcolor = data(Qt.BackgroundRole)
+                bgcolor = bgcolor.color().name() if bgcolor else 'transparent'
+                if bgcolor.lower() == '#ffffff':
+                    bgcolor = 'transparent'
+
+                font = data(Qt.FontRole)
+                weight = 'bold' if font and font.bold() else 'normal'
+
+                alignment = data(Qt.TextAlignmentRole) or Qt.AlignCenter
+                halign = ('left' if alignment | Qt.AlignLeft else
```
Comment (janezd) [inline Orange/canvas/report/report.py]: `&`, not `|`, I suppose? (Same below.)

File: Orange/canvas/report/report.py (hunk 2 above)
Comment (janezd) [inline Orange/canvas/report/report.py]: This is not exactly the same: the previous code would take `table.model()` disregarding the type. But I agree that checking the type is better than blindly accepting (any) `table.model()`. I suppose that the previous code was a result of (non-)linear evolution of the code. :)

**Comment by janezd** [comment]
Can you add a test for the bug described in https://github.com/biolab/orange3/pull/1309/files#r66053433?

**Comment by ajdapretnar** [comment]
@kernc @janezd From a user's standpoint this fix is ok. Can we sort out also the code so I can merge (or Janez can)? Thanks! :smile:

File: Orange/canvas/report/report.py (hunk 3 above)
Comment (kernc) [inline Orange/canvas/report/report.py]: Thanks.

**Comment by kernc** [comment]
@astaric Works, thanks!

@ajdapretnar @janezd Is done ... but is

Blocked on: https://github.com/biolab/orange3/pull/1300

**Comment by ajdapretnar** [comment]
@kernc #1300 seems to be merged now... Rebase and merge this one, too? Except that travis complains again...

**Pull request merged**